substrate/client/network/src/discovery.rs

```diff
@@ -1181,12 +1202,33 @@ mod tests {
 			&[kademlia_protocol_name(supported_genesis_hash, None)],
 			remote_addr.clone(),
 		);
+		// Note: legacy protocol is not supported without genesis hash and fork ID,
+		// if the legacy is the only protocol supported, then the peer will not be added.
 		discovery.add_self_reported_address(
 			&another_peer_id,
 			&[legacy_kademlia_protocol_name(&supported_protocol_id)],
 			another_addr.clone(),
 		);
```

Comment:
May be make it clear by renaming `another_peer_id` -> `unsupported_peer_id` and checking before and after the call that it's `remote_peer_id` that's added at this stage? We can still use `another_peer_id` below for the peer with both protocols.